lib/chef/file_access_control/windows.rb

```diff
@@ -90,11 +90,13 @@ def acls_equal(target_acl, actual_acl)
         target_acl.each do |target_ace|
           if target_ace.flags & INHERIT_ONLY_ACE == 0
             self_ace = target_ace.dup
-            self_ace.flags = 0
+            # We need flag value which is already being set in case of WRITE permissions as 3, so we will not be overwriting it with the hard coded value.
+            self_ace.flags = 0 unless target_ace.mask == Chef::ReservedNames::Win32::API::Security::WRITE
```

Comment:
I can't find `Security::WRITE` in `chef/win32/api/file.rb` - coudl this be intended to be GENERIC_WRITE, or FILE_GENERIC_WRITE?

Reply:
Is the intent to ensure that only the value '::WRITE' (3) is accepted? Or do we want to mask out any other options, so if `target_ace` has `flags` of `11`, we should retain the 3 and drop the rest: 

`self_ace.flags &= ...Security::WRITE`